### ISSUE TITLE
In some edit fields, Backspace fails to work with NVDA running, if NVDA's option "Use UIA with Microsoft Edge and other Chromium based browsers when available" is enabled

### ISSUE THREAD

**Issue by k-kolev1985** [nvaccess/nvda#12033]
### Steps to reproduce:
1. Launch NVDA.
2. From NVDA's advanced settings, enable the option "Use UIA with Microsoft Edge and other Chromium based browsers when available" (set it to "Yes").
4. Open Microsoft Edge and load the GitHub's "New issue" form page.
5. Go to the "Title" edit field and press ENTER on it to go into "focus mode".
6. Type a few words.
7. Press the Backspace key to delete the character on the left of the caret/cursor.

### Actual behavior:
NVDA plays its error sound and the character is not deleted.

### Expected behavior:
NVDA should allow the Backspace key to do its job as it should, as it does when "Use UIA with Microsoft Edge and other Chromium based browsers when available" is disabled (set to "No" or "Only when necessary").

### System configuration
- Operating system: Windows 10 Pro version 20H2 (build 19042.746), 64-bit, in Bulgarian with all locale settings set to "Bulgarian".
- NVDA version: alpha-21710,11ba984f, installed, in bulgarian.
- Microsoft Edge version: 88.0.705.56, 64-bit, in bulgarian.
- Processor: Intel Core i5-9400F at 2.90GHz.
- RAM Memory: 16.00GB.
- Graphics: MSI GeForce GTX-1050TI Gaming X 4G, 4096MB dedicated memory, desktop resolution set to 1920x1080.
- Sound Card: Realtek ALC892 at Intel Cannon Point PCH.

#### Does the issue still occur after restarting your computer?
Yes, it does.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No, because the feature that causes this issue is available only in alpha versions.

#### If addons are disabled, is your problem still occuring?
Yes, it is.

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
No, because in this case it is probably not related to this issue.

**Comment by gdata1** [comment]
Hello. I also noticed a problem in the editing field. I have when I get to a place where nothing is written nvda says a symbol with incomprehensible figures although there is nothing written. "Use UIA with Microsoft Edge and other chrome-based browsers included.

**Comment by Qchristensen** [comment]
This is still the case with NVDA 2023.2 Beta 2 under Windows 11 (64-bit) Version: 22H2, Build: 22621.2134 Using Microsoft Edge, Version: 115.0.1901.203.  It also happens when pressing control+backspace.  Here is a log fragment:

IO - inputCore.InputManager.executeGesture (11:42:49.022) - winInputHook (20812):
Input: kb(desktop):control+backspace
ERROR - scriptHandler.executeScript (11:42:49.032) - MainThread (15028):
error executing script: <bound method EditableText.script_caret_backspaceWord of <NVDAObjects.Dynamic_ChromiumUIAEditableTextWithAutoSelectDetectionUIA object at 0x0542F350>> with gesture 'ctrl+backspace'
Traceback (most recent call last):
  File "scriptHandler.pyc", line 295, in executeScript
  File "editableText.pyc", line 285, in script_caret_backspaceWord
  File "editableText.pyc", line 264, in _backspaceScriptHelper
  File "NVDAObjects\UIA\web.pyc", line 190, in move
  File "NVDAObjects\UIA\web.pyc", line 181, in _collapsedMove
  File "NVDAObjects\UIA\web.pyc", line 148, in _moveToEdgeOfReplacedContent
  File "baseObject.pyc", line 41, in __get__
  File "NVDAObjects\UIA\web.pyc", line 119, in _get_UIAElementAtStartWithReplacedContent
  File "monkeyPatches\comtypesMonkeyPatches.pyc", line 32, in __call__
_ctypes.COMError: (-2147467259, 'Unspecified error', (None, None, None, 0, None))


When use UIA in Edge is set to No or Default, you can backspace / control+backspace, it is only when it is enabled, and when trying to backspace from the last character in the edit field (ie, if you press left arrow once from the last character, you can then backspace / control+backspace and it works.

**Comment by Adriani90** [comment]
NVDA now deletes the character as expected after #16395 was merged, but NVDA is silent and does not report the characters deleted probably due to these errors:

```
IO - inputCore.InputManager.executeGesture (22:40:16.660) - winInputHook (20148):
Input: kb(laptop):backspace
ERROR - editableText.EditableText._backspaceScriptHelper (22:40:16.671) - MainThread (12092):
Error in testInfo.move
Traceback (most recent call last):
  File "editableText.pyc", line 266, in _backspaceScriptHelper
  File "NVDAObjects\UIA\web.pyc", line 190, in move
  File "NVDAObjects\UIA\web.pyc", line 181, in _collapsedMove
  File "NVDAObjects\UIA\web.pyc", line 148, in _moveToEdgeOfReplacedContent
  File "baseObject.pyc", line 41, in __get__
  File "NVDAObjects\UIA\web.pyc", line 119, in _get_UIAElementAtStartWithReplacedContent
_ctypes.COMError: (-2147467259, 'Unbekannter Fehler', (None, None, None, 0, None))
ERROR - editableText.EditableText.detectPossibleSelectionChange (22:40:16.696) - MainThread (12092):
Error in _updateSelectionAnchor
Traceback (most recent call last):
  File "editableText.pyc", line 405, in detectPossibleSelectionChange
  File "editableText.pyc", line 415, in _updateSelectionAnchor
  File "NVDAObjects\UIA\__init__.pyc", line 941, in compareEndPoints
_ctypes.COMError: (-2146233079, None, (None, None, None, 0, None))
```

This will also affect the HP browser.
